### PR TITLE
Feat: add reset backoff time in workflow status

### PR DESCRIPTION
**Pull request by FogDong**
Signed-off-by: FogDong <dongtianxin.tx@alibaba-inc.com>

### Description of your changes

add reset counter in workflow status

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### Special notes for your reviewer

/cc @Somefive

### CODE REVIEW

**Review by wonderflow — COMMENTED** [review]
what does this feature for?

File: apis/core.oam.dev/common/types.go [hunk 1]
```diff
@@ -327,9 +327,10 @@ type PolicyStatus struct {
 
 // WorkflowStatus record the status of workflow
 type WorkflowStatus struct {
-	AppRevision string       `json:"appRevision,omitempty"`
-	Mode        WorkflowMode `json:"mode"`
-	Message     string       `json:"message,omitempty"`
+	AppRevision  string       `json:"appRevision,omitempty"`
+	Mode         WorkflowMode `json:"mode"`
+	Message      string       `json:"message,omitempty"`
+	ResetCounter bool         `json:"resetCounter,omitempty"`
```
Comment (wonderflow) [inline apis/core.oam.dev/common/types.go:333]: how does user know the `Counter` mean? Does it mean loop backoff for users?

Comment (Somefive) [reply apis/core.oam.dev/common/types.go]: > how does user know the `Counter` mean? Does it mean loop backoff for users?

Yes. This field is designed to be used by Vela CLI to trigger immediate retry after long waiting. It is not necessary for user to understand it.

File: pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go [hunk 2]
```diff
@@ -313,8 +314,8 @@ func (r *Reconciler) gcResourceTrackers(logCtx monitorContext.Context, handler *
 		return r.result(r.patchStatus(logCtx, handler.app, phase)).requeue(baseGCBackoffWaitTime).ret()
 	}
 	logCtx.Info("GarbageCollected resourcetrackers")
-	if phase == common.ApplicationRendering {
-		return r.result(r.updateStatus(logCtx, handler.app, common.ApplicationRunningWorkflow)).ret()
+	if !usePatch {
+		return r.result(r.updateStatus(logCtx, handler.app, phase)).ret()
```
Comment (Somefive) [inline pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go:318]: Is there any potential changes here? @FogDong

**Review by Somefive — APPROVED** [review]
LGTM, thx!

**Review by barnettZQG — APPROVED** [review]

**Comment by Somefive** [comment]
Alternative change #3362 will be accepted.

**Pull request closed without merging**